src/runtime_src/core/tools/xbmgmt2/OO_UpdateBase.cpp

```diff
@@ -0,0 +1,752 @@
+#include "OO_UpdateBase.h"
+
+// XRT - Include Files
+#include "core/common/info_vmr.h"
+#include "core/common/message.h"
+#include "flash/flasher.h"
+#include "ReportPlatform.h"
+#include "tools/common/XBUtilitiesCore.h"
+#include "tools/common/ProgressBar.h"
+namespace XBU = XBUtilities;
+
+// 3rd Party Library - Include Files
+#include <boost/format.hpp>
+#include <boost/tokenizer.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+namespace po = boost::program_options;
+
+// System - Include Files
+#include <atomic>
+#include <chrono>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <thread>
+// =============================================================================
+
+// ----- H E L P E R M E T H O D S ------------------------------------------
+
+
+// ----- C L A S S   M E T H O D S -------------------------------------------
+
+OO_UpdateBase::OO_UpdateBase(const std::string &_longName, const std::string &_shortName, bool _isHidden )
+    : OptionOptions(_longName,
+                    _shortName,
+                    "Update base partition",
+                    boost::program_options::value<decltype(update)>(&update)->implicit_value("all")->required(),
+                    "Update the persistent images and/or the Satellite controller (SC) firmware image.  Valid values:\n"
+                      "  ALL   - All images will be updated\n"
+                      "  SHELL - Platform image\n"
+                      "  SC    - Satellite controller (Warning: Damage could occur to the device)\n"
+                      "  NO-BACKUP   - Backup boot remains unchanged",
+                    _isHidden)
+{
+  m_optionsDescription.add_options()
+    ("device,d", po::value<decltype(m_device)>(&m_device), "The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest")
+    ("image", boost::program_options::value<decltype(image)>(&image)->multitoken(),  "Specifies an image to use used to update the persistent device.  Valid values:\n"
+                                                                    "  Name (and path) to the mcs image on disk\n"
+                                                                    "  Name (and path) to the xsabin image on disk")
+    ("help", po::bool_switch(&m_help), "Help to use this sub-command")
+  ;
+
+  m_optionsHidden.add_options()
+    ("flash-type", boost::program_options::value<decltype(flashType)>(&flashType),
+      "Overrides the flash mode. Use with caution.  Valid values:\n"
+      "  ospi\n"
+      "  ospi_versal")
+    ;
+}
+
+// Create a list of images that are known to exist based on given names and paths
+static std::vector<std::string>
+find_flash_image_paths(const std::vector<std::string>& image_list)
+{
+  std::vector<std::string> path_list;
+  auto installedShells = firmwareImage::getIntalledDSAs();
+
+  for (const auto& img : image_list) {
+    // Check if the passed in image is absolute path
+    if (boost::filesystem::is_regular_file(img)){
```

Comment:
Use C++17 `<filesystem>` everywhere instead.

Reply:
Is there a difference between the two? This is a big change already and messing about with the filesystem may add an extra headache